common/requirements.h

```diff
@@ -32,6 +32,43 @@ class QString;
   TYPED_VECTOR_ITERATE(struct requirement, req_vec, preq)
 #define requirement_vector_iterate_end VECTOR_ITERATE_END
 
+/**
+ * A set of targets to evaluate requirements against. Depending on what the
+ * requirements in question are for, most of these entries will usually be
+ * nullptr. For instance, when evaluating the construction requirements for a
+ * building, there is no target unit, specialist etc.
+ *
+ * Fields should be added in ALPHABETICAL order since items need to be
+ * defined in the same order as they are declared and this will be easier to
+ * remember.
+ */
+struct req_context {
+  const struct action *action;
+  const struct impr_type *building;
+  const struct city *city;
+  const enum national_intelligence nintel = NI_COUNT;
+  const struct output_type *output;
+  const struct player *player;
+  const struct specialist *specialist;
+  const struct tile *tile;
+  const struct unit *unit;
+  const struct unit_type *utype;
+  const enum vision_layer vision_layer = V_COUNT;
+};
+
+// Context functions.
+#define req_action(_rc_) (_rc_ ? _rc_->action : nullptr)
```

Comment:
Macros, not functions. A function would be:
```suggestion
auto req_action(const req_context *rc) { return rc ? rc->action : nullptr; }
```
…and I'd like it better.